apps/system/test/unit/statusbar_test.js

```diff
@@ -1048,6 +1048,41 @@ suite('system/Statusbar', function() {
     });
   });
 
+  suite.only('Utility tray events should update once only', function() {
+    var dispatchEdgeSwipeEvent = function(event) {
+      window.dispatchEvent(new CustomEvent(event));
+    };
+
+    test('utility-tray-abortopen', function() {
+      dispatchEdgeSwipeEvent('utilitytraywillshow');
+      assert.isTrue(StatusBar.shouldUpdate('utility-tray-abortopen'));
+      assert.isFalse(StatusBar.shouldUpdate('utility-tray-abortopen'));
+    });
+
+    test('utility-tray-overlayopened', function() {
+      dispatchEdgeSwipeEvent('utilitytraywillshow');
+      assert.isTrue(StatusBar.shouldUpdate('utility-tray-overlayopened'));
+      assert.isFalse(StatusBar.shouldUpdate('utility-tray-overlayopened'));
+    });
+
+    test('utility-tray-abortclose', function() {
+      dispatchEdgeSwipeEvent('utilitytraywillhide');
+      assert.isTrue(StatusBar.shouldUpdate('utility-tray-abortclose'));
+      assert.isFalse(StatusBar.shouldUpdate('utility-tray-abortclose'));
+    });
+
+    test('utility-tray-overlayopened', function() {
+      dispatchEdgeSwipeEvent('utilitytraywillhide');
+      assert.isTrue(StatusBar.shouldUpdate('utility-tray-overlayclosed'));
+      assert.isFalse(StatusBar.shouldUpdate('utility-tray-overlayclosed'));
+    });
+
+    test('unknown event type should return true', function() {
+      assert.isTrue(StatusBar.shouldUpdate('xxx-unknown'));
+      assert.isTrue(StatusBar.shouldUpdate('xxx-unknown'));
+    });
+  });
+
```

Comment:
on the testing front
* we should update `testEventThatPause` / `testEventThatResumeIfNeeded` to check that the `evt.type` is properly passed as a parameter
* it's a bit "internal" to test `shouldUpdate`, and `testEventThatPause` / `testEventThatResumeIfNeeded` are already making sure the all the events are taking care of, what we need is to test the causeForPause mechanism.
* and to do that "externally", well, it's hard. but I'll let a spy on `_updateIconVisibility` pass since we already have a lot :) so the causeForPause test should do something like
```
dispatch a `sheets-gesture-begin`
dispatch an `iconshown` event and assert that `_updateIconVisibility` isn't called
dispatch another `sheets-gesture-begin`
dispatch a `sheets-gesture-end`
dispatch an `iconshown` event and assert that `_updateIconVisibility` **is** called
```

Since testEventThatPause / testEventThatResumeIfNeeded is covering that we call pause/resume properly, we can cover the pause/resume mechanism with only one "example", the `sheet-gesture-*` events.

* and we'll also need to add a test (well 2 tests) covering that `resumeUpdate()` only calls `_updateIconVisibility` if `!this.isPause()`